libsolidity/formal/SMTEncoder.cpp

```diff
@@ -69,7 +69,15 @@ bool SMTEncoder::analyze(SourceUnit const& _source)
 				analysis = false;
 			}
 
-	return analysis;
+	if (!analysis)
+		return false;
+
+	set<FunctionCall const*> abiCalls = collectABICalls(&_source);
+	for (auto const& source: _source.referencedSourceUnits(true))
+		abiCalls += collectABICalls(source);
+	m_context.state().buildABIFunctions(abiCalls);
```

Comment:
Any reason not just to move this part up before the loop above? Especially since you loop across the referenced source units there already anyways?

Reply:
And then remove the loop? Otherwise the pointer would always be reassigned inside `buildABIFunctions`